tests/api.py

```diff
@@ -0,0 +1,170 @@
+import sys
+import cdis_oauth2client
+import sheepdog
+
+from flask import Flask, jsonify
+from flask.ext.cors import CORS
+from flask_sqlalchemy_session import flask_scoped_session
+from psqlgraph import PsqlGraphDriver
+
+from cdis_oauth2client import OAuth2Client, OAuth2Error
+from cdisutils.log import get_handler
+from indexclient.client import IndexClient as SignpostClient
+from userdatamodel.driver import SQLAlchemyDriver
+
+from sheepdog.auth import AuthDriver
+from sheepdog.errors import APIError, setup_default_handlers, UnhealthyCheck
+from sheepdog.version_data import VERSION, COMMIT, DICTVERSION, DICTCOMMIT
+
+
+# recursion depth is increased for complex graph traversals
+sys.setrecursionlimit(10000)
+DEFAULT_ASYNC_WORKERS = 8
+
+def app_register_blueprints(app):
+    # TODO: (jsm) deprecate the index endpoints on the root path,
+    # these are currently duplicated under /index (the ultimate
+    # path) for migration
+    v0 = '/v0'
+    app.url_map.strict_slashes = False
+
+    app.register_blueprint(cdis_oauth2client.blueprint, url_prefix=v0+'/oauth2')
+
+def async_pool_init(app):
+    """Create and start an pool of workers for async tasks."""
+    n_async_workers = (
+        app.config
+        .get('ASYNC', {})
+        .get('N_WORKERS', DEFAULT_ASYNC_WORKERS)
+    )
+    app.async_pool = sheepdog.utils.scheduling.AsyncPool()
+    app.async_pool.start(n_async_workers)
+
+
+def db_init(app):
```

Comment:
I'm sort of suspicious that most of the stuff set up here (`oauth2` client, `signpost` client, etc.) should just be mocked out to keep the tests and services atomic. In fact, are either of those clients used in the testing?
 
(Also see `patch_indexclient` in `utils.py`.)